src/main/java/app/ultratechies/api/AppUsers/UserService.java

```diff
@@ -43,51 +47,52 @@ public void deleteUser(Long userId) {
-                            Optional<AppUser> userByUsername= userRepository.findAppUserByUsername(username);
-                            if (userByUsername.isPresent()){
-                                throw new IllegalStateException("username already exists!");
-                            }
-                            appUser.setUsername(username);
-                        }
-
-                        if (name!= null &&
-                                name.length()>0 &&
-                                !Objects.equals(appUser.getName(),name)){
-
-                            appUser.setName(name);
-                        }
-
-                        if (email!= null &&
-                                email.length()>0 &&
-                                !Objects.equals(appUser.getEmail(),email)){
-                            Optional<AppUser> userByEmail = userRepository.findAppUserByEmail(email);
-                            if (userByEmail.isPresent()) {
-                                throw new IllegalStateException("email taken!");
-                            }
-                            appUser.setEmail(email);
-                        }
-
-                        if (password!= null &&
-                                password.length()>0 &&
-                                !Objects.equals(appUser.getPassword(),password)){
-
-                            appUser.setPassword(password);
-                        }
-
-                        if (photo!= null &&
-                                photo.length()>0 &&
-                                !Objects.equals(appUser.getPhoto(),photo)){
-
-                            appUser.setPhoto(photo);
-                        }
-    }
+        AppUser appUser = userRepository.findById(userId)
+                .orElseThrow(() -> new IllegalStateException("user with userId: " + userId + " does not exist!"));
+
+        if (username != null &&
+                username.length() > 0 &&
+                !Objects.equals(appUser.getUsername(), username)) {
+            Optional<AppUser> userByUsername = userRepository.findAppUserByUsername(username);
+            if (userByUsername.isPresent()) {
+                throw new IllegalStateException("User with username " + username + " already exists!");
+            } else {
+                appUser.setUsername(username);
+            }
+        }
+
+        if (name != null &&
+                name.length() > 0 &&
+                !Objects.equals(appUser.getName(), name)) {
 
+            appUser.setName(name);
+        }
+
+        if (email != null &&
+                email.length() > 0 &&
+                !Objects.equals(appUser.getEmail(), email)) {
+            Optional<AppUser> userByEmail = userRepository.findAppUserByEmail(email);
+            if (userByEmail.isPresent()) {
+                throw new IllegalStateException("User with email " + appUser.getEmail() + " already exists");
```

Comment:
@iannjari the error message here should be `IllegalStateException("User with email " + email + " already exists")`

Reply:
Noted